src/coreclr/vm/multicorejit.cpp

```diff
@@ -1185,10 +1209,13 @@ void MulticoreJitManager::StartProfile(AppDomain * pDomain, ICLRPrivBinder *pBin
 
     if ((pProfile != NULL) && (pProfile[0] != 0)) // Ignore empty file name, just same as StopProfile
     {
+        int dontGatherProfile = (int)CLRConfig::GetConfigValue(CLRConfig::INTERNAL_MultiCoreJitNoProfileGather);
```

Comment:
There is already an existing check to prevent recording info, based on `m_fRecorderActive` below. Instead of most of the new checks for null arrays, it would be better to avoid setting `m_fRecorderActive` to `true` below when recording is disabled through the new config. Most of the new checks for null arrays can instead just be asserts. `WriteOutput` I believe would be the only place that the check for null arrays would be necessary.

Reply:
I tried this approach at first, but there are problems with it, because `m_fRecorderActive` is not checked everywhere. For example, `MulticoreJitManager::RecordMethodJitOrLoad` doesn't check it before calling `MulticoreJitRecorder::RecordMethodJitOrLoad`. Also, `MulticoreJitRecorder::RecordMethodJitOrLoad` can be called from `MulticoreJitManager::RequestMethodCode`, which also doesn't check for `m_fRecorderActive`. Do you mean that I should instead add checks for `m_fRecorderActive` everywhere, where they are required?

Reply:
I think `m_fRecorderActive` should should be checked everywhere before recording. If there are missing places they should probably be added. Some of the checks are through `MulticoreJitManager::IsRecorderActive`.

From what I can tell these are the caller chains to `MulticoreJitRecorder::RecordMethodJitOrLoad`:
- `MulticoreJitManager::RecordMethodJitOrLoad`
  - `MethodDesc::GetPrecompiledCode` checks `MulticoreJitManager::IsRecorderActive`
  - `MethodDesc::JitCompileCodeLocked` checks `MulticoreJitManager::IsRecorderActive`
- `MulticoreJitRecorder::RequestMethodCode` checks `MulticoreJitManager::IsRecorderActive`

Reply:
Ah, I see, I've missed `IsRecorderActive` for some reason. Thanks! 
I'll recheck again with m_fRecorderActive set to false.